### DISCUSSION TITLE
Refresh Token - Auth for Tesla - when new?

### DISCUSSION THREAD

**Discussion by pedroplusrodrigues** [alandtse/tesla#704] [Q&A]
I've got everything nicely running. Since a month or so. I use "Auth for Tesla" and on the iOS App there is a timer that counts from 7h 59m.

My question: When do i need a new "refresh token"? Is there somewhere a timer or sort of? Does it refreshes itself on HomeAssistant?

Hope the question is not tooo stupid :-)

**Comment by pedroplusrodrigues** [comment]
Is there an entity that informs you when the tesla integration needs an new token?

**Reply by alandtse** [reply]
No. It'll just prompt you when it needs it.

**Comment by pedroplusrodrigues** [comment]
Ok, thanks!
